From 4e17c050dac8f2c6e2d278c4c4a27001c8d7d164 Mon Sep 17 00:00:00 2001
From: Jakub Hrozek <jhrozek@redhat.com>
Date: Thu, 17 Nov 2016 15:43:54 +0100
Subject: [PATCH 57/79] TESTS: move helper fixtures to back up and restore a
 file to a utility module
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The fixtures will be useful for tests that set up and restore a user and
group database. While it would be possible to import them already, the
functions were previously used in a test and importing from a test seems
a bit like a hack.

Reviewed-by: Pavel Březina <pbrezina@redhat.com>
Reviewed-by: Lukáš Slebodník <lslebodn@redhat.com>
---
 src/tests/intg/Makefile.am |  1 +
 src/tests/intg/ent_test.py | 14 --------------
 src/tests/intg/util.py     | 14 ++++++++++++++
 3 files changed, 15 insertions(+), 14 deletions(-)

diff --git a/src/tests/intg/Makefile.am b/src/tests/intg/Makefile.am
index e81e5ee950d1a1e50ff478a34ae29853a9783c8b..90e0f308043aa89da5afc0fd353f7d373ac6425f 100644
--- a/src/tests/intg/Makefile.am
+++ b/src/tests/intg/Makefile.am
@@ -1,5 +1,6 @@
 dist_noinst_DATA = \
     config.py.m4 \
+    util.py \
     sssd_id.py \
     sssd_ldb.py \
     sssd_netgroup.py \
diff --git a/src/tests/intg/ent_test.py b/src/tests/intg/ent_test.py
index 598930324a41ad9ff473e94e7f2deb302e1bc942..6b240ae9682331be7f4485a1ac81fad4cebde5e6 100644
--- a/src/tests/intg/ent_test.py
+++ b/src/tests/intg/ent_test.py
@@ -19,25 +19,11 @@
 import re
 import os
 import io
-import shutil
 import pytest
 import ent
 from util import *
 
 
-def backup_envvar_file(name):
-    path = os.environ[name]
-    backup_path = path + ".bak"
-    shutil.copyfile(path, backup_path)
-    return path
-
-
-def restore_envvar_file(name):
-    path = os.environ[name]
-    backup_path = path + ".bak"
-    os.rename(backup_path, path)
-
-
 @pytest.fixture(scope="module")
 def passwd_path(request):
     name = "NSS_WRAPPER_PASSWD"
diff --git a/src/tests/intg/util.py b/src/tests/intg/util.py
index 66ec0baa1aea4f8f9a3210d33640f087c80c6862..2b40311bd1560e05f3aafb34136a1b0efc8a1b49 100644
--- a/src/tests/intg/util.py
+++ b/src/tests/intg/util.py
@@ -21,6 +21,7 @@ import re
 import os
 import subprocess
 import config
+import shutil
 
 UNINDENT_RE = re.compile("^ +", re.MULTILINE)
 
@@ -64,3 +65,16 @@ def first_dir(*args):
     for arg in args:
         if os.path.isdir(arg):
             return arg
+
+
+def backup_envvar_file(name):
+    path = os.environ[name]
+    backup_path = path + ".bak"
+    shutil.copyfile(path, backup_path)
+    return path
+
+
+def restore_envvar_file(name):
+    path = os.environ[name]
+    backup_path = path + ".bak"
+    os.rename(backup_path, path)
-- 
2.9.3

